src/components/App/App.jsx

```diff
@@ -0,0 +1,116 @@
+import Navbar from "../Navbar/Navbar"
+import Home from "../Home/Home"
+import Profile from "../Profile/Profile"
+import Recipes from "../Recipes/Recipes"
+import Pantry from "../Pantry/Pantry"
+import ShoppingCart from "../ShoppingCart/ShoppingCart"
+import './App.css';
+
+import { BrowserRouter, Routes, Route } from "react-router-dom"
+import { useState, useEffect } from "react"
+
+export default function App() {
+
+  var basicProducts = [
+    {name: "apple",
+    quantity: 2},
+    {name: "orange",
+    quantity: 5},
+    {name: "grape",
+    quantity: 8},
+    {name: "bacon",
+    quantity: 3},
+    {name: "milk",
+    quantity: 2},
+    {name: "pasta",
+    quantity: 1}
+  ]
+
+  var basicProductForm = {
+    name: "",
+    quantity: ""
+  }
+
+const [products, setProducts] = useState(basicProducts)
+const [productForm, setProductForm] = useState(basicProductForm)
+
+const handleProductQuantity = (productName, operation) => {
+  // find index of productName
+  let itemIndex = products.findIndex(item => item.name === productName)
+
+  let newProducts = []
+  for (let i = 0; i < products.length; i++) {
+    newProducts.push(products[i])
+  }
+
+  if (operation == "add") {
+    newProducts[itemIndex].quantity += 1
+  } 
+  else if (operation === "minus") {
+    newProducts[itemIndex].quantity -= 1
+    if (newProducts[itemIndex].quantity == 0) {
+      newProducts.splice(itemIndex, 1)
+    }
+  } else {
+    newProducts.splice(itemIndex, 1)
```

Comment:
I don't understand why you're removing the item if the operation is not "add" or "minus". Is there another operation that I'm missing and if so can you add that in a comment?

Edit: I see the last action is remove. Can you still add something here to indicate that?

Reply:
I'll change it to an `else if` and throw an error for `else`